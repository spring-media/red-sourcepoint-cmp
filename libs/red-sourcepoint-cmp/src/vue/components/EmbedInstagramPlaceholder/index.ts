export { default as EmbedInstagramPlaceholder } from './EmbedInstagramPlaceholder.vue';
