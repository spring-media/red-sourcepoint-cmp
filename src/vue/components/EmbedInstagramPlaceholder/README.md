# Placeholder component for Instagram embeds

<p>
  <img src="../../../../docs/embed-instagram-placeholder.png" alt="Embed placeholder Instagram" width="500" />
</p>

<details>
<summary>Example</summary>

```vue
<template>
  <embed-instagram-placeholder :privacyManagerId="privacyManagerId"></embed-instagram-placeholder>
</template>

<script>
import { EmbedInstagramPlaceholder } from '@spring-media/red-sourcepoint-cmp/dist/esm/vue/components/EmbedInstagramPlaceholder';

export default {
  components: { EmbedInstagramPlaceholder },
  data: () => ({
    privacyManagerId: 12345,
  }),
};
</script>

<style lang="scss">
@import '~@spring-media/red-sourcepoint-cmp/dist/esm/vue/components/EmbedInstagramPlaceholder.css';
</style>
```
</details>

## Props

| Name             | Type   | Required | Description |
| ---------------- | ------ | -------- | ----------- |
| privacyManagerId | number | true     | Id of a privacy manager to open when clicking on certain areas within the placeholder |
