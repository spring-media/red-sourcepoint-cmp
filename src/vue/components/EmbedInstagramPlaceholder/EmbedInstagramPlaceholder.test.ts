import { mount } from '@vue/test-utils';
import EmbedInstagramPlaceholder from './EmbedInstagramPlaceholder.vue';

describe('EmbedPlaceholderInstagram', () => {
  it('should render without any errors', () => {
    expect(mount(EmbedInstagramPlaceholder).element).toMatchSnapshot();
  });
});
