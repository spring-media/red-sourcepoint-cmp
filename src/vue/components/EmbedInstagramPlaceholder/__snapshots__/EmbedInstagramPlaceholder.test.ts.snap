// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmbedInstagramPlaceholder should render without any errors 1`] = `
<div
  class="embed-placeholder__container"
>
  <div
    class="embed-placeholder__header"
  >
    <svg
      fill="none"
      height="312"
      width="100%"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="12"
        cy="12"
        fill="#E9ECEF"
        r="12"
      />
       
      <rect
        fill="#E9ECEF"
        height="10"
        width="60"
        x="32"
        y="7"
      />
       
      <rect
        fill="#E9ECEF"
        height="calc(100% - 32)"
        width="100%"
        y="32"
      />
       
      <svg
        x="calc(50% - 20)"
        y="50%"
      >
        <path
          d="M11.6001 0H28.4001C34.8001 0 40.0001 5.2 40.0001 11.6V28.4C40.0001 31.4765 38.778 34.427 36.6026 36.6024C34.4271 38.7779 31.4766 40 28.4001 40H11.6001C5.20012 40 0.00012207 34.8 0.00012207 28.4V11.6C0.00012207 8.52349 1.22226 5.57298 3.39768 3.39756C5.57311 1.22214 8.52361 0 11.6001 0ZM11.2001 4C9.29056 4 7.45922 4.75857 6.10895 6.10883C4.75869 7.45909 4.00012 9.29044 4.00012 11.2V28.8C4.00012 32.78 7.22012 36 11.2001 36H28.8001C30.7097 36 32.541 35.2414 33.8913 33.8912C35.2416 32.5409 36.0001 30.7096 36.0001 28.8V11.2C36.0001 7.22 32.7801 4 28.8001 4H11.2001ZM30.5001 7C31.1632 7 31.799 7.26339 32.2679 7.73223C32.7367 8.20107 33.0001 8.83696 33.0001 9.5C33.0001 10.163 32.7367 10.7989 32.2679 11.2678C31.799 11.7366 31.1632 12 30.5001 12C29.8371 12 29.2012 11.7366 28.7324 11.2678C28.2635 10.7989 28.0001 10.163 28.0001 9.5C28.0001 8.83696 28.2635 8.20107 28.7324 7.73223C29.2012 7.26339 29.8371 7 30.5001 7ZM20.0001 10C22.6523 10 25.1958 11.0536 27.0712 12.9289C28.9466 14.8043 30.0001 17.3478 30.0001 20C30.0001 22.6522 28.9466 25.1957 27.0712 27.0711C25.1958 28.9464 22.6523 30 20.0001 30C17.348 30 14.8044 28.9464 12.9291 27.0711C11.0537 25.1957 10.0001 22.6522 10.0001 20C10.0001 17.3478 11.0537 14.8043 12.9291 12.9289C14.8044 11.0536 17.348 10 20.0001 10ZM20.0001 14C18.4088 14 16.8827 14.6321 15.7575 15.7574C14.6323 16.8826 14.0001 18.4087 14.0001 20C14.0001 21.5913 14.6323 23.1174 15.7575 24.2426C16.8827 25.3679 18.4088 26 20.0001 26C21.5914 26 23.1175 25.3679 24.2428 24.2426C25.368 23.1174 26.0001 21.5913 26.0001 20C26.0001 18.4087 25.368 16.8826 24.2428 15.7574C23.1175 14.6321 21.5914 14 20.0001 14Z"
          fill="#868E96"
        />
      </svg>
       
      <svg
        x="calc(100% - 20)"
      >
        <path
          d="M5.80006 0H14.2001C17.4001 0 20.0001 2.6 20.0001 5.8V14.2C20.0001 15.7383 19.389 17.2135 18.3013 18.3012C17.2136 19.3889 15.7383 20 14.2001 20H5.80006C2.60006 20 6.10352e-05 17.4 6.10352e-05 14.2V5.8C6.10352e-05 4.26174 0.611131 2.78649 1.69884 1.69878C2.78655 0.61107 4.26181 0 5.80006 0ZM5.60006 2C4.64528 2 3.72961 2.37928 3.05448 3.05442C2.37935 3.72955 2.00006 4.64522 2.00006 5.6V14.4C2.00006 16.39 3.61006 18 5.60006 18H14.4001C15.3548 18 16.2705 17.6207 16.9456 16.9456C17.6208 16.2705 18.0001 15.3548 18.0001 14.4V5.6C18.0001 3.61 16.3901 2 14.4001 2H5.60006ZM15.2501 3.5C15.5816 3.5 15.8995 3.6317 16.1339 3.86612C16.3684 4.10054 16.5001 4.41848 16.5001 4.75C16.5001 5.08152 16.3684 5.39946 16.1339 5.63388C15.8995 5.8683 15.5816 6 15.2501 6C14.9185 6 14.6006 5.8683 14.3662 5.63388C14.1318 5.39946 14.0001 5.08152 14.0001 4.75C14.0001 4.41848 14.1318 4.10054 14.3662 3.86612C14.6006 3.6317 14.9185 3.5 15.2501 3.5ZM10.0001 5C11.3261 5 12.5979 5.52678 13.5356 6.46447C14.4733 7.40215 15.0001 8.67392 15.0001 10C15.0001 11.3261 14.4733 12.5979 13.5356 13.5355C12.5979 14.4732 11.3261 15 10.0001 15C8.67398 15 7.40221 14.4732 6.46453 13.5355C5.52684 12.5979 5.00006 11.3261 5.00006 10C5.00006 8.67392 5.52684 7.40215 6.46453 6.46447C7.40221 5.52678 8.67398 5 10.0001 5ZM10.0001 7C9.20441 7 8.44135 7.31607 7.87874 7.87868C7.31613 8.44129 7.00006 9.20435 7.00006 10C7.00006 10.7956 7.31613 11.5587 7.87874 12.1213C8.44135 12.6839 9.20441 13 10.0001 13C10.7957 13 11.5588 12.6839 12.1214 12.1213C12.684 11.5587 13.0001 10.7956 13.0001 10C13.0001 9.20435 12.684 8.44129 12.1214 7.87868C11.5588 7.31607 10.7957 7 10.0001 7Z"
          fill="black"
        />
      </svg>
    </svg>
  </div>
   
  <div
    class="embed-placeholder__headline"
  >
    
    An dieser Stelle findest du Inhalte aus Instagram
  
  </div>
   
  <div
    class="embed-placeholder__description"
  >
    
    Um mit Inhalten aus Instagram und anderen Sozialen Netzwerken zu interagieren oder diese darzustellen, brauchen
    wir deine Zustimmung.
  
  </div>
   
  <button
    class="embed-placeholder__button"
  >
    
        Soziale Netzwerke aktivieren
      
  </button>
   
  <div
    class="embed-placeholder__footer-text"
  >
    
        Ich bin damit einverstanden, dass mir externe Inhalte aus Sozialen Netzwerken angezeigt werden. Damit können
        personenbezogene Daten an Drittanbieter übermittelt werden. Mehr dazu findest du in der
        
    <a
      class="embed-placeholder__text-link embed-placeholder__link-description"
      href="#"
      rel="noopener"
      target="_blank"
    >
      Beschreibung dieses Datenverarbeitungszweck
    </a>
    
        und in den Datenschutzinformationen dieser
        
    <a
      class="embed-placeholder__text-link embed-placeholder__link-vendor-list"
      href="#"
      rel="noopener"
      target="_blank"
    >
      Drittanbieter
    </a>
    .
      
  </div>
</div>
`;
